### DISCUSSION TITLE
clang-tidy error clang-diagnostic-error

### DISCUSSION THREAD

**Discussion by h1laryz** [clangd/clangd#1939] [Q&A] [answered]
Hi, I am using gcc/g++ as compilers in my projects, but also I want to use .clang-tidy as static analyzer.
While building I am getting an error 

error: unknown argument '-static-libasan'; did you mean '-static-libsan'? [clang-diagnostic-error]
14068 warnings and 1 error generated.

How can I disable this warning in .clang-tidy?
Here is my .clang-tidy file:
Checks: >
  bugprone-*,
  cppcoreguidelines-*,
  modernize-*,
  performance-*,
  -cppcoreguidelines-avoid-magic-numbers,
  -cppcoreguidelines-pro-bounds-array-to-pointer-decay,
  -cppcoreguidelines-pro-type-reinterpret-cast,
  -cppcoreguidelines-pro-type-vararg,
  -modernize-use-trailing-return-type,
  readability-magic-numbers,
  google-runtime-int,
  misc-*,
  clang-analyzer-*,
  -clang-diagnostic-unknown-argument,
  -no-unknown-warning-option,
  -unknown-argument,

WarningsAsErrors: '*'
HeaderFilterRegex: 'a^'
FormatStyle: file

**Comment by HighCommander4** [comment]
Sorry, I'm a bit confused. Are you using clangd? 

> While building I am getting an error
> 
> error: unknown argument '-static-libasan'; did you mean '-static-libsan'? [clang-diagnostic-error] 14068 warnings and 1 error generated.

What program / command is giving you this error?

**Reply by h1laryz** [reply]
set_target_properties(
        ${PROJECT_NAME} PROPERTIES
        CXX_CLANG_TIDY "${CLANG_TIDY_EXECUTABLE}"
)

this from CMakeLists.txt getting an error during compilation. Because clang-tidy says it is an error. 
Clangd I am using for vim/vscode and it also says that this -static-libasan flag is unknown, but actually this flag comes from a third_party library.

Now I think about 2 ways:
1) somehow in clang-tidy disable this warning
2) use clang as compiler instead of gcc/g++

**Reply by HighCommander4** [reply]
> Clangd I am using for vim/vscode and it also says that this -static-libasan flag is unknown, but actually this flag comes from a third_party library.

Could you share [clangd logs](https://clangd.llvm.org/troubleshooting.html#gathering-logs)?

**Reply by h1laryz** [reply]
![изображение](https://github.com/clangd/clangd/assets/71406854/de5ee225-d49e-44d9-b83e-014521ee075c)

**Reply by h1laryz** [reply]
this flag goes from third party library named "userver" it checks if compiler is not clang and sets this compile flag. 
So, how can i disable it in clang-tidy checks?
My clang-tidy file you can fine above, what should I add there?
Thanks so much

**Reply by HighCommander4** [reply, marked as answer]
Ok, I see, it's a clang driver diagnostic about a gcc command line argument that clang doesn't recognize.

So, for clangd, you can fix it by creating a `.clangd` file containing:

```yaml
CompileFlags:
  Remove: [-static-libasan]
```

For clang-tidy, I don't know if there is anything comparable; please ask at https://github.com/llvm/llvm-project/issues.

**Reply by h1laryz** [reply]
Created this issue https://github.com/llvm/llvm-project/issues/82314
